src/codegen/compile.ml

```diff
@@ -8302,62 +8348,18 @@ and compile_prim_invocation (env : E.t) ae p es at =
 
   (* Cycles *)
   | SystemCyclesBalancePrim, [] ->
-    SR.Vanilla,
-    Stack.with_words env "dst" 4l (fun get_dst ->
-      get_dst ^^
-      IC.cycle_balance env ^^
-      get_dst ^^
-      Cycles.load_cycles env
-    )
+    SR.Vanilla, Cycles.balance env
   | SystemCyclesAddPrim, [e1] ->
-    SR.unit,
-    let (set_cycles, get_cycles) = new_local env "cycles" in
-    compile_exp_vanilla env ae e1 ^^
-    set_cycles ^^
-    get_cycles ^^
-    Cycles.guard env ^^
-    get_cycles ^^
-    Cycles.push_high env ^^
-    get_cycles ^^
-    Cycles.push_low env ^^
-    IC.cycles_add env
+    SR.unit, compile_exp_vanilla env ae e1 ^^ Cycles.add env
   | SystemCyclesAcceptPrim, [e1] ->
-    SR.Vanilla,
-    let (set_cycles, get_cycles) = new_local env "cycles" in
-    Stack.with_words env "dst" 4l (fun get_dst ->
-      compile_exp_vanilla env ae e1 ^^
-      set_cycles ^^
-      get_cycles ^^
-      Cycles.guard env ^^
-      get_cycles ^^
-      Cycles.push_high env ^^
-      get_cycles ^^
-      Cycles.push_low env ^^
-      get_dst ^^
-      IC.cycles_accept env ^^
-      get_dst ^^
-      Cycles.load_cycles env
-    )
+    SR.Vanilla, compile_exp_vanilla env ae e1 ^^ Cycles.accept env
   | SystemCyclesAvailablePrim, [] ->
-    SR.Vanilla,
-    Stack.with_words env "dst" 4l (fun get_dst ->
-      get_dst ^^
-      IC.cycles_available env ^^
-      get_dst ^^
-      Cycles.load_cycles env
-    )
+    SR.Vanilla, Cycles.available env
   | SystemCyclesRefundedPrim, [] ->
-    SR.Vanilla,
-    Stack.with_words env "dst" 4l (fun get_dst ->
-      get_dst ^^
-      IC.cycles_refunded env ^^
-      get_dst ^^
-      Cycles.load_cycles env
-    )
+    SR.Vanilla, Cycles.refunded env
+
   | SetCertifiedData, [e1] ->
-    SR.unit,
-    compile_exp_vanilla env ae e1 ^^
-    IC.set_certified_data env
+    SR.unit, compile_exp_vanilla env ae e1 ^^ IC.set_certified_data env
```

Comment:
was this intentional?

Reply:
Yes, a bit more consistent with the lines above. Irrelevant cosmetics, though